tensorflow_datasets/core/download/kaggle.py

```diff
@@ -46,10 +46,32 @@
 class KaggleFile(object):
   """Represents a Kaggle competition file."""
   _URL_PREFIX = "kaggle://"
+  _DATASET_PREFIX = "dataset"
+  _COMPETITION_PREFIX = "competition"
 
-  def __init__(self, competition_name, filename):
+  def __init__(self, competition_name, filename, download_type=None):
     self._competition_name = competition_name
     self._filename = filename
+    self._type = download_type
+  
+  @staticmethod
+  def get_type(
+      competition_name,
+      dataset_value=_DATASET_PREFIX,
+      competition_value=_COMPETITION_PREFIX):
+
+    return  dataset_value if "/" in competition_name \
+          else competition_value
+  
+  def get_original_url(self):
+    return "%s%s/%s" % (self._URL_PREFIX,
+                        self._competition_name,
+                        self._filename) 
+  @property
+  def type(self):
+    if self._type is None:
+      self._type = KaggleFile.get_type(self._competition_name)
```

Comment:
If the type can be deduced from the competition name, why not compute it in the `__init__`, rather than in here ?
Also if the type can be deduced from the competition name, why is the extra `download_type` required ?